### PR TITLE
Document martian 4.0 features.

### PR DESCRIPTION
**Pull request by adam-azarchs**

### CODE REVIEW

File: content/writing-pipelines/index.md [hunk 1]
```diff
@@ -115,9 +115,117 @@ files for [Vim](https://www.vim.org/), [Atom](https://atom.io/),
...
+already assigned in the input call explicitly (e.g. `foo` in the example).
+
+#### Collection types
+
+Martian also supports collections of values as arrays or typed maps.  These are
+declared using a syntax that is familar to users of C-style languages.  Arrays
+are declared as for example `int[]`.  Typed maps (available in the martian 4.0
+preview) always have string keys, and
+are declared as for example `map<int>`.  These can be combined as for example
+`map<int[][]>[]`.
```
Comment (sreenathkrishnan) [inline content/writing-pipelines/index.md:199]: Cool, I din't know this. Is 2d array the highest supported by martian?

File: content/writing-pipelines/index.md [hunk 2]
```diff
@@ -246,17 +397,24 @@ stage SORT_ITEMS(
...
-`mrf` takes a list of MRO filenames as arguments. By default, it will output the formatted code back to `stdout`. If given the `--rewrite` option, it will write the formatted code back into the original files. If given the `--all` option, it will rewrite all MRO files found in your `MROPATH`. For consistency of your MRO codebase, consider configuring editor save-hooks or git commit-hooks that run `mrf --rewrite` or `mrf --all`.
+`mrf` takes a list of MRO filenames as arguments. By default, it will output
+the formatted code back to `stdout`. If given the `--rewrite` option, it will
+write the formatted code back into the original files. If given the `--all`
+option, it will rewrite all MRO files found in your `MROPATH`. For consistency
+of your MRO codebase, consider configuring editor save-hooks or git
+commit-hooks that run `mrf --rewrite` or `mrf --all`.
 
-`mrf` does not support any arguments that affect the formatting, otherwise it would not be canonical!
+`mrf` does not support any arguments that affect the formatting, otherwise it
```
Comment (sreenathkrishnan) [inline content/writing-pipelines/index.md:416]: Would be good to mention `--includes` here

File: content/writing-pipelines/index.md (hunk 1 above)
Comment (adam-azarchs) [reply content/writing-pipelines/index.md:199]: Array dimension is a signed 16-bit integer.  Please don't push the limits on that range :-)

File: content/writing-pipelines/index.md (hunk 2 above)
Comment (adam-azarchs) [reply content/writing-pipelines/index.md:416]: Done.

**Review by sreenathkrishnan — APPROVED** [review]
LGTM! Thanks!

**Review by nlhepler — APPROVED** [review]
:+1:

**Pull request merged**